lib/mempool/mempool.js

```diff
@@ -282,6 +282,27 @@ class Mempool extends EventEmitter {
         this.untrackClaim(entry);
     }
 
+    // Remove all GooSig based airdrops from the mempool
+    // when the block at the height that disables GooSig
+    // is added to the mempool. It would be wasteful to do
+    // this check on every block after GooSig is disabled,
+    // so only do it once on the block height that disables
+    // GooSig.
```

Comment:
By directly accessing `this.network.airdrop`, it bypasses the whole deployment state abstraction, which no other parts of the codebase do that. I'm considering removing the checks below on `if (... this.network)` and just using `if (await this.hasGooBye())` instead. This would mean that all future blocks past GooSig being disabled would cause all of the airdrops in the mempool to be checked. While it is wasteful, its less code and more simple. The check could then technically be deleted from the codebase far after GooSig is disabled, or we could add another deployment that is used by the mempool to stop it from checking. It may also be possible to do something like:

```
if (!this.nextState.hasGooBye() && (await this.getNextState()).hasGooBye())
```

This logic would be similar to https://github.com/handshake-org/hsd/pull/305/files#diff-17fec30dbf2ab21611334d7aaa3aec43R668, basically comparing the current state and the next state.

Reply:
Actually you bring up a good point here. `GooBye` is NOT a deployment -- which is a term that usually refers to soft-forks. It's a hard consensus rule, set before the project launches. It's more akin to `network.auctionStart` and for that reason I think you can pull out all the `hasGoobye()` stuff and just compare to the constant flag-height.

Reply:
I agree with @pinheadmz. I think we should move all goosig related logic out of the deployment state and reference the network parameters where necessary. However, I like that the network params are split into a boolean representing whether goosig should be disabled and a height specifying when (if necessary).

Reply:
This has been updated with hardcoded constants. I also removed the logline `GooSig has been disabled`, I think that logline would still be nice to have but no blocker.